spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -2722,7 +2721,11 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
 
       case op
           if isCometSink(op) && op.output.forall(a =>
-            supportedDataType(a.dataType, allowComplex = true)) =>
+            supportedDataType(
+              a.dataType,
+              allowComplex =
+                usingDataFusionParquetExec(conf) || CometConf.COMET_CONVERT_FROM_PARQUET_ENABLED
```

Comment:
Why `COMET_CONVERT_FROM_*` is allowed only here but not other places?

Reply:
This allows complex types for native reader and if conversions enabled. For other cases complex types would fall back to Spark behavior.

@parthchandra @mbutrovich @andygrove please have a second look if this condition is okay or may be improved

Reply:
This check seems fine. Could could get a complex type in a CometSink if either of these conditions is true.